aptos-move/framework/aptos-framework/sources/object.spec.move

```diff
@@ -475,7 +475,12 @@ spec aptos_framework::object {
         aborts_if !global<ObjectCore>(object_address).allow_ungated_transfer;
     }
 
-    spec burn<T: key>(owner: &signer, object: Object<T>) {
+    spec burn<T: key>(_owner: &signer, _object: Object<T>) {
+        pragma aborts_if_is_partial;
+        aborts_with EBURN_NOT_ALLOWED;
```

Comment:
I think we can write `aborts_if true;` instead. Specifying error codes in aborts statements is nice to have, but it hasn't been done much in the framework so far.

Reply:
This essentially is the same though right?  Just more specific?

Reply:
You can also write `aborts_if CONDITION with ERROR_CODE`, where `ERROR_CODE` represents the category rather than a specific error reason. This pattern was commonly used in the Diem Framework, but hasn't been adopted in Aptos (https://github.com/aptos-labs/aptos-core/blob/8b4d7e2527c9c0d4c1a034d6bda37712804f668e/third_party/move/move-examples/diem-framework/move-packages/DPN/sources/AccountAdministrationScripts.move#L61)